Improve documentation of json parser functions

idoqo opened the pull request:
Signed-off-by: Michael Okoko <okokomichaels@outlook.com>

<!--
Thank you for contributing to the MariaDB Server repository!

You can help us review your changes faster by filling this template <3

If you have any questions related to MariaDB or you just want to
hang out and meet other community members, please join us on
https://mariadb.zulipchat.com/ .
-->

<!--
If you've already identified a https://jira.mariadb.org/ issue
that seems to track this bug/feature, please add its number below.
-->
<!--
An amazing description should answer some questions like:
1. What problem is the patch trying to solve?
2. If some output changed, what was it looking like before
   the change and how it's looking with this patch applied
3. Do you think this patch might introduce side-effects in
   other parts of the server?
-->
## Description
I had to work with some of the JSON parser functions while working on MDEV-21130. They were helpful but also, the comments could be improved. This tries to make it clearer e.g., 
- by describing how to get desired values after calling `json_get_array_item`

KartikSoneji reviewed (changes requested):
The functions are called `json_*` instead of `js_*` everywhere else.

Can you please fix this line as well? GitHub doesn't allow changing lines that haven't been touched by the original PR.
https://github.com/MariaDB/server/blob/274534d82af9aae851f4a3c64aea7c03648b988c/include/mysql/service_json.h#L41

cvicentiu commented:
Hi @idoqo!

Can you please rebase your changes on top of 10.4 branch, as that's where the typos were initially introduced. Also, please squash the 2 commits into 1.

I'll merge the PR once you've done that.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MariaDB/server?pullRequest=1876) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/MariaDB/server?pullRequest=1876) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: idoqo<br/>:white_check_mark: nayuta-yanagisawa<br/>:x: vaintroub<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MariaDB/server?pullRequest=1876) it.</sub>

idoqo commented:
I've rebased and changed the base branch to 10.4 - though not sure why AppVeyor reports it as non-mergeable at the moment (git doesn't seem to complain of conflicts).
